contracts/examples/ping-pong-egld/src/lib.rs

```diff
@@ -38,6 +38,11 @@ pub trait PingPong {
 			"smart contract not active yet"
 		);
 
+		require!(
+			self.get_block_nonce() < self.get_deadline(),
+			"smart contract already ended"
```

Comment:
"smart contract already ended" doesn't sound too good. Maybe rephrase to "deadline has passed" or something similar.